app/api/main/controller/company_controller.py

```diff
@@ -54,4 +54,20 @@ async def full_evaluation(search_request: SearchRequest) -> EvaluatedCompanyDTO:
     except Exception as e:
         raise HTTPException(
             status_code=500, detail=str(e)
-        )
\ No newline at end of file
+        )
+    
+
+@router.patch("/company/compliance")
+async def toggle_compliance(request: SearchRequest):
+    print(f"Received request to toggle compliance for: {request.company_name}")
+    try:
+        updated_company = await CompanyService.toggle_compliance(request.company_name)
+        if not updated_company:
+            raise HTTPException(
+                status_code=404, detail=f"Company '{request.company_name}' not found."
+            )
+        return {"message": f"Compliance for '{request.company_name}' updated successfully.", "updated_company": updated_company}
```

Comment:
This logic doesn't make sense since we don't check if the company exists after updating it. After bringing company exists logic into the controller, this logic can be removed and you can call the return the company directly.

That is,

```suggestion
        return updated_company
```

Reply:
Returning anything here is redundant

Reply:
Agreed, fixed in another comment.